src/expr/src/scalar/func.rs

```diff
@@ -3659,31 +3659,36 @@ fn pad_leading<'a>(
     Ok(Datum::String(temp_storage.push_string(buf)))
 }
 
-fn substr<'a>(datums: &[Datum<'a>]) -> Datum<'a> {
-    let string: &'a str = datums[0].unwrap_str();
-    let mut chars = string.chars();
+fn substr<'a>(datums: &[Datum<'a>]) -> Result<Datum<'a>, EvalError> {
+    let s: &'a str = datums[0].unwrap_str();
 
-    let start_in_chars = datums[1].unwrap_int64() - 1;
-    let mut start_in_bytes = 0;
-    for _ in 0..cmp::max(start_in_chars, 0) {
-        start_in_bytes += chars.next().map(|char| char.len_utf8()).unwrap_or(0);
-    }
+    let raw_start_idx = datums[1].unwrap_int64() - 1;
+    let start_idx = std::cmp::max(raw_start_idx, 0) as usize;
```

Comment:
I think you might want to change this to `usize::try_from(std::cmp::max(raw_start_idx, 0))`.

Reply:
Ah, good call.